jms/src/main/scala/akka/stream/alpakka/jms/Jms.scala

```diff
@@ -32,9 +32,17 @@ sealed trait JmsSettings {
   def acknowledgeMode: Option[AcknowledgeMode]
 }
 
+object Destination {
+  def createQueue(name: String)(session: jms.Session) = session.createQueue(name)
+  def createTopic(name: String)(session: jms.Session) = session.createTopic(name)
+}
 sealed trait Destination
```

Comment:
You could add `createDestination` in the `Destination` trait and make the subclasses implement it. And add a third sub-class where the factory method is user-defined.
This would remove the need to pattern match in `JmsConnector`.